### PR TITLE
Fix incorrect handling of Elimination and Case schemes

### LINKED ISSUE #10816: `Scheme Case` and `Scheme Elimination` do not match `Set Case Analysis Schemes`
#### Description of the problem

One would think that `Scheme Case` would generate the same sort of lemma as [`Set Case Analysis Schemes`](https://coq.inria.fr/refman/user-extensions/proof-schemes.html#coq:flag.case-analysis-schemes), but instead it seems to generate a scheme identical to the `Minimality` one, except named `case` instead of `rec`.

Also, it should be possible to generate both the dependent and the non-dependent case analysis schemes.

See Also #7958

#### Coq Version

8.9.0

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #10816 

depends 
- https://github.com/coq/coq/pull/19016

Overlays:
- https://github.com/HoTT/Coq-HoTT/pull/1958 (backwards compatible)
- https://github.com/mattam82/Coq-Equations/pull/594 (backwards compatible)

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: vernac/indschemes.ml [hunk 1]
```diff
@@ -361,25 +361,25 @@ let smart_ind qid =
...
-  | (None, ({sch_type; sch_qualid; sch_sort} as sch)) :: q
-   -> let ind = smart_ind sch_qualid in
-      let sort_of_ind = Indrec.pseudo_sort_family_for_elim ind (snd (Inductive.lookup_mind_specif env ind)) in
-      let suffix = scheme_suffix_gen sch sort_of_ind in
-      let newid = Nameops.add_suffix (Nametab.basename_of_global (Names.GlobRef.IndRef ind)) suffix in
-      let newref = CAst.make newid in
-      (newref, sch_isdep sch_type, ind, sch_sort) :: name_and_process_schemes env q
-
-let do_mutual_induction_scheme ?(force_mutual=false) env l =
+(* Resolve the name of a scheme using an enviornment and extract some
```
Comment (herbelin) [inline vernac/indschemes.ml:364]: enviornment -> environment

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
CI failure unrelated, let @coqbot merge now

**Pull request merged**